### PR TITLE
Stop hardcoding properties in newConvertToARMFunctionBuilder

### LINKED ISSUE #3549: Don't hardcode properties in newConvertToARMFunctionBuilder
**Describe the current behavior**
`v2/tools/generator/internal/armconversion/convert_to_arm_function_builder.go` has the following code snippet

```
	var emptyCollectionProperties set.Set[string]
	if forceEmptyCollections {
		// TODO: These should not be hardcoded here, instead should be in the azure-arm.yaml config
		emptyCollectionProperties = set.Make(
			"Tags",
			"NodeLabels",
			"NodeTaints")
	}
```

**Describe the improvement**
The properties should be specified in the configuration file, not in code.

### PR DESCRIPTION
**Pull request by matthchr**
This fixes #3549.

This uses a property tag for a few reasons:
  1. Some types are flattened and other approaches are easily lost when flattening occurs. Putting the tag onto the property preserves it even through flattening.
  2. In many ways this behavior is like an augmented `json:omitempty`, so it (IMO) makes sense to have a tag just like for JSON. It makes it clearer when looking at the Go object that the serialization behavior of these fields is special.

This also fixes an issue with the configuration where hierarchical configurations didn't propagate the checks for usage.

**How does this PR make you feel**:
gif

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

File: v2/tools/generator/internal/config/object_model_configuration.go [hunk 1]
```diff
@@ -62,8 +62,15 @@ func NewObjectModelConfiguration() *ObjectModelConfiguration {
 	}
 
 	// Initialize group access fields here (alphabetical, please)
```
Comment (super-harsh) [inline v2/tools/generator/internal/config/object_model_configuration.go:64]: Do we still need this comment here?

Comment (super-harsh) [inline v2/tools/generator/internal/config/object_model_configuration.go:64]: Do we still need this comment here?

File: v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go [hunk 2]
```diff
@@ -231,20 +237,13 @@ func (c *armConversionApplier) addARMConversionInterface(
...
-
 	addInterfaceHandler := func(t *astmodel.ObjectType) (astmodel.Type, error) {
 		result := t.WithInterface(armconversion.NewARMConversionImplementation(
 			armDef.Name(),
 			objectType,
+			kubeDef.Name(),
 			c.idFactory,
-			typeType,
-			payloadType))
+			typeType))
```
Comment (super-harsh) [inline v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go:246]: Should we consider renaming `typeType` to `typeKind`?

**Review by super-harsh — APPROVED** [review]

File: v2/tools/generator/internal/config/object_model_configuration.go (hunk 1 above)
Comment (matthchr) [reply v2/tools/generator/internal/config/object_model_configuration.go:64]: I think so?

File: v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go (hunk 2 above)
Comment (matthchr) [reply v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go:246]: Good catch - done

File: v2/tools/generator/internal/config/object_model_configuration.go (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/config/object_model_configuration.go:64]: Yes, as a guidepost for future maintainers that _here_ is where group access is configured.

**Pull request merged**